osquery/filesystem/linux/proc.cpp

```diff
@@ -16,54 +16,165 @@
 #include <osquery/filesystem.h>
 #include <osquery/logger.h>
 
+#include "osquery/core/conversions.h"
+#include "osquery/filesystem/linux/proc.h"
+
 namespace osquery {
+const char* kLinuxProcPath = "/proc";
 
-const std::string kLinuxProcPath = "/proc";
+Status procGetProcessNamespaces(const std::string& process_id,
+                                ProcessNamespaceList& namespace_list) {
+  namespace_list.clear();
 
-Status procProcesses(std::set<std::string>& processes) {
-  // Iterate over each process-like directory in proc.
-  boost::filesystem::directory_iterator it(kLinuxProcPath), end;
-  try {
-    for (; it != end; ++it) {
-      if (boost::filesystem::is_directory(it->status())) {
-        // See #792: std::regex is incomplete until GCC 4.9
-        if (std::atoll(it->path().leaf().string().c_str()) > 0) {
-          processes.insert(it->path().leaf().string());
-        }
-      }
+  for (std::string namespace_name :
+       {"cgroup", "ipc", "mnt", "net", "pid", "user", "uts"}) {
```

Comment:
Can you bring this set into a constant defined above, so it's easier to extend in the future?